indispensable-asn1/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Exception.kt

```diff
@@ -19,4 +19,18 @@ class Asn1OidException(message: String, val oid: ObjectIdentifier) : Asn1Excepti
  * Runs [block] inside [catching] and encapsulates any thrown exception in an [Asn1Exception] unless it already is one
  */
 @Throws(Asn1Exception::class)
-inline fun <reified R> runRethrowing(block: () -> R) = catchingUnwrappedAs(::Asn1Exception, block).getOrThrow()
\ No newline at end of file
+inline fun <reified R> runRethrowing(block: () -> R) = catchingUnwrappedAs(::Asn1Exception, block).getOrThrow()
+
+/**
+ * Decodes this ASN.1 structure using the provided [decoder] lambda, rethrowing any caught exception
+ * as an [Asn1Exception].
+ * This is a wrapper around [Asn1Structure.decodeAs] that ensures exceptions thrown during decoding are
+ * consistently rethrown as [Asn1Exception], using the [runRethrowing] utility.
+ */
+inline fun <reified R> Asn1Structure.decodeRethrowing(
```

Comment:
shouldn't this be `internal`? or perhaps moving it to the internals module

Reply:
i think not. if you are going to define your own external asn.1 structures, this is really handy